### ISSUE TITLE
Table().Where(Expression) gives Member access failed to compile expression exception

### ISSUE THREAD

**Issue by Stark-92** [praeclarum/sqlite-net#535]
Hey,

I am using praeclarum sqlite-net library in xamarin PCL library, when I am using :

var item = sQLiteConnection.Table<MyClass>().Where(e =>  (e.Id == CurrentId) && (e.name == Currentname));

It gives exception: Member access failed to compile expression

CurrentId and Currentname both have a value.
The table does exist with its columns.

MyClass is:

public class MyClass 
{

    [Ignore]
    public List<ClassB> Bs{ get; set; }

    public string BsAsJson { get; set; }

    public string Datestamp { get; set; }

    [PrimaryKey]

    public string ClassId { get; set; }

    public string SomeId { get; set; }

    public string name { get; set; }

    public MyClass ()
    {

    }

    public MyClass (string datestamp, string id)
    {
        Bs= new List<ClassB>();

        Datestamp = datestamp;

        ClassId = id;

        name= "Bagera";

        SomeId= "SomeValue";

    }

}

I have asked about this problem in stack over flow: 
http://stackoverflow.com/questions/43784932/praeclarum-sqlite-net-table-whereexpression-gives-member-access-failed-to-co

And in Xamarin forums:

https://forums.xamarin.com/discussion/94718/praeclarum-sqlite-net-table-where-expression-gives-exception

**Comment by Stark-92** [comment]
Passing CurrentId as parameter fixed my problem.

Before, I used the function like this where Current Id = StaticClassA.MyStaticList[StaticIndex].Id

    `public list<MyClass> GettAll()
{
  var item = sQLiteConnection.Table<MyClass>().Where(e =>  (e.Id ==  CurrentId) && (e.name == Currentname));

  return item.ToList();
}`
The code before will throw => exception: Member access failed to compile expression.

But, when I pass CurrentId as parameter it works fine:

`public list<MyClass> GettAll(string id)
{
  var item = sQLiteConnection.Table<MyClass>().Where(e =>  (e.Id == id) && (e.name == Currentname));

  return item.ToList();
}
`
I still don't know what is the difference !

Hope this will help someone.

**Comment by praeclarum** [comment]
Yes the problem is with calling methods from the query. I will look into supporting this.

**Comment by suchithm** [comment]
Worked! Thanks for the update

**Comment by joehanna** [comment]
Workaround: add `.ToList()` to the Table result first. Try:

```
 var item = sQLiteConnection.Table<MyClass>().ToList().Where(e =>  (e.Id ==  CurrentId) && (e.name == Currentname));
```

**Comment by Astolfoho** [comment]
@joehanna this is not a good solution, because when you use ToList() all the table will be load to memory before Where filter.

Sorry for my bad english.

**Comment by joehanna** [comment]
I understand @astolfoho. That is why I called it a workaround and not a solution. Work arounds are not usually ideal and are only temporary until the issue is resolved.

**Comment by otvoiski** [comment]
I transformed the table in the list, it worked for me.

**Comment by zoli13** [comment]
Any news on this?
I have to same issue sql.Table<table>().Where( xxxx) --> "System.NotSupportedException: Member access failed to compile expression"

**Comment by leeuwk004** [comment]
Are there any plans on resolving this issue?

**Comment by whSwitching** [comment]
same problem when `connection.Table<MyTable>().Where(a => !a.Property.HasValue || ids.Contains(a.Property.Value)).ToArray()`
MyTable.Property  is Nullable<int>

**Comment by ldcx9** [comment]
Thask,It help me.
